### PR TITLE
OverlayManager refactor and Flash

### PR DESCRIPTION
**Pull request by rneuser**
* Refactors OverlayManager to be more versatile. You can now add multiple overlays to clients, as well as send parameters along for a specific overlay if desired.

* Adds the flash item from SS13

* MeleeWeaponComponent.Attack now returns when OnHitEntities returns false

### CODE REVIEW

File: Resources/Textures/Objects/Melee/flash.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+    "version": 1,
+    "size": {
+        "x": 32,
+        "y": 32
+    },
+    "license": "GNU AGPL v3",
```
Comment (ShadowCommander) [inline Resources/Textures/Objects/Melee/flash.rsi/meta.json:7]: This is a code license, you'll want the find the content license which for SS13 is usually CC BY-SA of some sort.

**Comment by rneuser** [comment]
Well so I changed it to a draft PR because I'm not sure this is really the best way to go about this so lemme know. Some code is definitely in non-prime locations, but that's just a minor issue. 

Essentially now instead of the OverlayManager being able to only set one overlay at a time, I changed it to be able to handle multiple overlays that can have their own state that is sent to the client.

Big thing that's probably not ideal is the IConfigurable interface, which is implemented on the actual Overlay class to be able to apply state changes. Might be a bit too generic? idk

File: Content.Server/GameObjects/Components/Weapon/Melee/FlashComponent.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            foreach (var entity in entities)
+            {
+                Flash(entity);
+            }
+
+            return true;
+        }
+
+        public bool UseEntity(UseEntityEventArgs eventArgs)
```
Comment (ShadowCommander) [inline Content.Server/GameObjects/Components/Weapon/Melee/FlashComponent.cs:60]: Make sure to remove this

File: Content.Shared/GameObjects/Components/Mobs/SharedOverlayEffectsComponent.cs [hunk 3]
```diff
@@ -13,21 +21,70 @@ public abstract class SharedOverlayEffectsComponent : Component
...
 
-    public enum ScreenEffects
+    [Serializable, NetSerializable]
+    public class OverlayContainer
     {
-        None,
-        CircleMask,
-        GradientCircleMask,
+        [ViewVariables(VVAccess.ReadOnly)]
+        public string ID { get; }
```
Comment (ShadowCommander) [inline Content.Shared/GameObjects/Components/Mobs/SharedOverlayEffectsComponent.cs:28]: I'm not sure it matters, but you might be able to use an enum here instead.

Comment (rneuser) [reply Content.Shared/GameObjects/Components/Mobs/SharedOverlayEffectsComponent.cs:28]: I think I will add an enum to Content.Shared so the server and client can agree on overlay ID's properly. Caveat being when someone adds an overlay they have to also add an entry to the enum. I'll probably add a test or something to check for that.

File: Content.Server/GameObjects/Components/Weapon/Melee/FlashComponent.cs (hunk 2 above)
Comment (rneuser) [reply Content.Server/GameObjects/Components/Weapon/Melee/FlashComponent.cs:94]: This is now AoE like the ss13 flash

**Pull request merged**